src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/Implementation/FrameworkReferenceAssemblyAttachedCollectionSourceProvider.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.ComponentModel;
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Internal.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections.Implementation
+{
+    [Export(typeof(IAttachedCollectionSourceProvider))]
+    [Name(nameof(FrameworkReferenceAssemblyAttachedCollectionSourceProvider))]
+    [VisualStudio.Utilities.Order(Before = HierarchyItemsProviderNames.Contains)]
+    internal sealed class FrameworkReferenceAssemblyAttachedCollectionSourceProvider : DependenciesAttachedCollectionSourceProviderBase
+    {
+        private readonly IRelationProvider _relationProvider;
+
+        [ImportingConstructor]
+        public FrameworkReferenceAssemblyAttachedCollectionSourceProvider(IRelationProvider relationProvider)
+            : base(DependencyTreeFlags.FrameworkDependency)
+        {
+            _relationProvider = relationProvider;
+        }
+
+        protected override bool TryCreateCollectionSource(
+            IVsHierarchyItem hierarchyItem,
+            string flagsString,
+            string? target,
+            IRelationProvider relationProvider,
+            [NotNullWhen(returnValue: true)] out AggregateRelationCollectionSource? containsCollectionSource)
+        {
+            if (ErrorHandler.Succeeded(hierarchyItem.HierarchyIdentity.Hierarchy.GetProperty(
+                hierarchyItem.HierarchyIdentity.ItemID, (int)__VSHPROPID.VSHPROPID_BrowseObject, out object browseObject)))
+            {
+                ICustomTypeDescriptor? desc = browseObject as ICustomTypeDescriptor;
```

Comment:
You shouldn't be doing this cast, this is an implementation detail of the browse object. You should be just using ProjectItem.Properties.

Reply:
How do I get a `ProjectItem` for a `IVsHierarchyItem`?

Reply:
https://github.com/dotnet/project-system/pull/6156